### DISCUSSION TITLE
Search for Orders/Decisions by Judge By Date

### DISCUSSION THREAD

**Discussion by pistorinoj** [freelawproject/courtlistener#2822] [General]
Newbie here, so be gentle.

I am doing some research regarding compliance with the Civil Justice Reform Act (CJRA) where, e.g., reports are prepared showing motions pending more than 6 months (with cut-offs dates of March 31 and September 30 each year).

Some judges issue decisions on or near these cut-off dates, while others take steps that mean pending motions will not be reported.

I would like to be able to search by judge for all orders, etc., issued within a particular date range.

There is no way to do this using PACER itself, that I am aware of.

Is there someway to do this within RECAP or the data it has?

Thanks for any help.

Regards,
James

**Comment by mlissner** [comment]
Hi James, can you share some of the searches you've tried, as URLs? This seems fairly doable either via the API or the front end. The challenges will be that:

1. We don't have all the data in PACER, so all our results have that as a caveat.

2. it's difficult to identify that an item is an opinion or order. We do have a boolean for whether something is free in PACER (indicating that the judge or clerk marked it as an order), but various jurisdictions do a terrible job flagging things correctly, so this boolean usually isn't much help. Without that to guide you, you have to play with the query strings to see if you can figure things out. (The [Advanced Search page](https://www.courtlistener.com/help/search-operators/) can help refine that query.)

3. Date ranges are easy to search.

**Reply by pistorinoj** [reply]
Not sure what you mean by sharing a URL.

For the examples I will use here, assume that I am searching the District of Columbia District Court, Judge Walton, want both open and closed cases, any case type, and want court orders/opinions issued between March 1 and April 1, for a particular year.

Using PACER, if you specifically go to the District of Columbia, you can select all cases (open and closed) but can only specify a date range of the LAST ENTRY and cannot select a particular judge.  Thus, this search is underinclusive, overinclusive, and does not result in a list of orders/opinions.  Instead, you get a list of all cases that satisfy the criteria (thus including cases by judges other than Walton).  You do not get a list of orders/opinions.  

Further, these are only the cases where the LAST ENTRY was between those dates.  I am interested in orders/opinions issued within the specified date range, regardless of whether there are later entries in the same case.  We are now several months from March and it is possible that there have been more entries and, therefore, items will not be returned that I am looking for.

Not sure if this helps but here is a URL that reports the results of the above search.
https://ecf.dcd.uscourts.gov/cgi-bin/iquery.pl?192065383443958-L_1_0-1

In my ideal world, I could get a list across all cases, whether open or not, or orders/opinions issued by Walton within a specified date range.

While I have not used the PACER or RECAP API, I can do C++/Python programming.
So, if I got results in a CSV file that would be easy.
With a bit more work, I could learn the API.
The PACER API does not appear to allow the specific search I am describing.

Thanks for any help.

Regards,
James

**Reply by mlissner** [reply]
> Not sure what you mean by sharing a URL.

I mean after you've done a search in the RECAP Archive that doesn't work, can you share that URL so we know what you've tried?

Send that over, and we can discuss that. In the meantime, if you have a CL username, send that too, please, because I think you're going to need special API access to pull this off.

**Reply by pistorinoj** [reply]
OK.  I did a search using CL for District of Columbia, Walton, March 1, 2023-April 1, 2023, any status, etc.
That said there were no results.
Here is the URL in CL
https://www.courtlistener.com/?type=o&q=dateFiled%3A%5B2023-03-01T00%3A00%3A00z%20TO%202023-04-01T00%3A00%3A00z%5D&type=o&order_by=dateFiled%20desc&judge=Walton&stat_Precedential=on&stat_Non-Precedential=on&stat_Errata=on&stat_Separate%20Opinion=on&stat_In-chambers=on&stat_Relating-to%20orders=on&stat_Unknown%20Status=on&court=dcd

I did the same thing searching the RECAP archive and also got zero results.
Here is the URL
https://www.courtlistener.com/?type=o&q=dateFiled%3A%5B2023-03-01T00%3A00%3A00z%20TO%202023-04-01T00%3A00%3A00z%5D&type=o&order_by=dateFiled%20desc&judge=Walton&stat_Precedential=on&stat_Non-Precedential=on&stat_Errata=on&stat_Separate%20Opinion=on&stat_In-chambers=on&stat_Relating-to%20orders=on&stat_Unknown%20Status=on&court=dcd

I know that there are results in this range.  For example, if you look at DC case #18-cv-00038, this is a case assigned to Walton and that he issued an order on March 31, 2023.  I have a list of 13 other cases that should be returned.

I need a search because I want to run the same exercise for September 2022, March 2022, etc.

My CL username is: pistorinoj.

Thanks for all your help.

Regards,
James

**Reply by mlissner** [reply]
Well, this isn't going to please anybody, but you need to capitalize the Z's in your timestamps:

https://www.courtlistener.com/?q=dateFiled%3A%5B2023-03-01T00%3A00%3A00Z%20TO%202023-04-01T00%3A00%3A00Z%5D&type=r&order_by=score%20desc&court=dcd

Sorry. I hate this syntax and we need to make it less horrible for humans. 

So that should help. 

The other thing: You're going to want more API access to do your work, so I've granted that to your account.

**Comment by pistorinoj** [comment]
Thanks Mike.

I used lowercase "z"s because that is what your example shows (at least in
Firefox).
https://www.courtlistener.com/help/search-operators/
Note: when I cut and paste this into an email, it changes the "z" to
uppercase but that is not how it shows on your web page.

Running that search, you are getting 450+ cases with nearly 8,000 entries.
This is because you are getting every docket entry that happened in that
one month period.
The simple filing of a case typically will result in 5+ docket entries.
By contrast, I am looking for activity by the Court itself.

Before I dive into learning/programming an API interface, I wonder if there
is a way to include a search for the filer of a docket entry?
Every docket entry appears to have a filer.
The ones for Walton himself, as opposed to the clerk's office, list the
filer as "lcrbw3".
Thus, is there a way I can add a requirement that the result must include
"lcrbw3"?

BTW, I see that you are/were in the Berkeley area.  I am in Menlo Park.

Thanks for your help.

Regards,
James

**Reply by mlissner** [reply]
> when I cut and paste this into an email, it changes the "z" to
> uppercase

I guess this could be a font issue, where the font looks like a lowercase to you. Want to share a screenshot? It looks fine to me.

> is a way to include a search for the filer of a docket entry?

Yeah, maybe, if we have that info (which we don't always). You could try searching for the filer's name in the docket entry text by using the `description:` field.

See if that helps?

**Comment by pistorinoj** [comment]
Here is a partial screenshot of the Advanced Query Techniques and Operators webpage.
![2023-06-20_16-37-40](https://github.com/freelawproject/courtlistener/assets/5090980/ee73a7a3-0bcb-49cc-942d-db79d7eefe53)

I will try a search as you suggest and see what I get.

Thanks again,
James

**Reply by mlissner** [reply]
OK, yep, that's not great. Mine looks uppercase. Can you see which font is doing that? Maybe we can tweak it, but probably we just need to tweak the docs.

**Comment by pistorinoj** [comment]
I am using Firefox with the defaults (Times New Roman, 16 point font).

**Comment by edrobinson** [comment]
From "Advanced Query Techniques and Operators"

Date Queries
Date queries require the [ISO-8601 standard](https://en.wikipedia.org/wiki/ISO_8601) date formatting. This means that dates must be formatted as follows:

YYYY-MM-DD

In English that's year-month-day. For example, here's a date range that finds all docket filings from October 2018:

[dateFiled:[2018-10-01 TO 2018-10-31]](https://www.courtlistener.com/?type=r&q=dateFiled%3A%5B2018-10-01%20TO%202018-10-31%5D)

This is  just a bit confusing . . .

**Reply by mlissner** [reply]
Happy to take suggestions? Do you think this is a documentation thing or something that we should change in how it works?

**Reply by edrobinson** [reply]
I can't really make suggestions at this point and this is certainly not a criticism of the API or the docs. It's just a comment.

***About:***
I have been doing development since Assembler. I have not done much Python but can read it OK. I see some C# and JS on GitHub but I am using PHP at the moment. It really is a good language . . .

***What's up***:
I have been trying to get my head around the CL APIs for a month or so and find the documentation kind of thin. All of the examples in the docs like to use curl. I don't care for it myself but it helps a but with the URLs. I have done quite a bit with SOAP and a few other APIs
so the CL APIs read easily.

It looks like I have to pick some starting point and drill down with several requests. 

I ran this search using the site's advance search:
https://www.courtlistener.com/?type=o&q=&type=o&order_by=score%20desc&citation=1%20idaho%2033&court=idd%20idb%20idaho%20idahoctapp

It returned 2 cases on screen one of which i was interested in.

I copied the URL and will work with it. and so far no good. I am using the Request which works a lot like PY request

Thanks for listening.
Ed
